### PR TITLE
Streaming v0.2.1

### PR DESCRIPTION
**Pull request by vchiley**
~~updates examples/llm dataset / dataloader to use [Streaming v0.2.1](https://github.com/mosaicml/streaming/releases/tag/v0.2.1)~~
waiting for Streaming v0.2.2

### CODE REVIEW

File: llm/main.py [hunk 1]
```diff
@@ -118,10 +118,12 @@ def build_composer_model(cfg):
 
 
 def build_dataloader(cfg, device_batch_size):
-    if cfg.name == 'c4':
-        return build_c4_dataloader(cfg, device_batch_size)
-    else:
-        raise ValueError(f'Not sure how to build model with name={cfg.name}')
+    return build_text_dataloader(cfg, device_batch_size)
```
Comment (abhi-mosaic) [inline llm/main.py:121]: Extra line here?

**Review by abhi-mosaic — APPROVED** [review]
Looks good! I think we should do some E2E tests with wandb plots just to confirm that things are working... the `num_canonical_nodes=8` thing will definitely affect things a bit, I'm expecting that startup time on 1 physical node (from trainer init to first batch) will be a bit longer than before, but I don't know by how much. Basically we have to wait for 8 shards to download rather than 1. WDYT @vchiley ?

**Review by growlix — APPROVED** [review]
LGMT!

**Comment by abhi-mosaic** [comment]
I think we are still waiting on some stuff here, but wanted to add one more potential blocker:


> [CO-1577](https://mosaicml.atlassian.net/browse/CO-1577): [dataloader test will xfail](https://github.com/vchiley/examples/blob/dataset020/llm/tests/test_dataloader.py#L20); it fails when used on a single GPU when not launched with a torch.distributed.launch cmd


Does this mean that the instructions in the README, for testing your local dataset will not work?
https://github.com/mosaicml/examples/tree/main/llm#test-the-dataloader

**Comment by abhi-mosaic** [comment]
Gonna close this for now and move over to #61

**Pull request closed without merging**